activation/nipost_test.go

```diff
@@ -807,6 +807,344 @@ func TestNIPSTBuilder_PoetUnstable(t *testing.T) {
 	})
 }
 
+// TestNIPoSTBuilder_PoETConfigChange checks if
+// it properly detects added/deleted PoET services and re-registers if needed.
+func TestNIPoSTBuilder_PoETConfigChange(t *testing.T) {
+	t.Parallel()
+
+	sig, err := signing.NewEdSigner()
+	require.NoError(t, err)
+
+	challenge := types.NIPostChallenge{
+		PublishEpoch: postGenesisEpoch + 2,
+	}
+
+	const (
+		poetProverAddr  = "http://localhost:9999"
+		poetProverAddr2 = "http://localhost:9988"
+	)
+
+	t.Run("1 poet deleted BEFORE round started -> continue get proof from known poets", func(t *testing.T) {
```

Comment:
Disregarding existing registrations should work up to the round _end_
```suggestion
	t.Run("1 poet deleted BEFORE round ended -> continue get proof from known poets", func(t *testing.T) {
```

Reply:
I meant this deadline https://github.com/spacemeshos/go-spacemesh/blob/develop/activation/nipost.go#L239

So scenario should illustrate case, when:
- registration was submitted to multiple poets in time
- user deletes 1 poet and restart the node
- node restarted, but PoET round in current round didn't started, so we will just continue with existing poet registration, because it was added to multiple poets

That's why test name is `1 poet deleted BEFORE round ended`

Reply:
OK, but then you also need a test when a poet is removed from the config while the round is _in progress_.